src/Page.js

```diff
@@ -983,6 +984,28 @@ class Page {
    * Collect page content inserted by plugins
    */
   collectPluginsAssets(content) {
+    const getResolvedAssetElement = (html, tagName, attrName, plugin, pluginName) => {
```

Comment:
only one suggestion:

The implementation of `utils.buildScript` and `buildStyleSheet` plus this `getResolvedElement` seems rather redundant, where build* produces `<script src =...></script>`, and getResolvedElement later reparses it. If buildScript could instead return something like

```js
{ type: 'scriptsrc', src: 'file.js' }
```

then we wouldn't need to use cheerio at all, just convert the src first and then return  `<script src =...></script>`.

Reply:
One con with this suggestion would be that the user cannot manually return `<script src = path></script>`.

Maybe it's better to leave it in so the user has more choices?

Reply:
Reworked the `buildScript/buildStylesheet` a little with the new `pluginUtils` from #1100, its the same from a plugin author's standpoint though ( changes are limited to the first of the 4 commits ).

```
{ type: 'scriptsrc', src: 'file.js' }
```
for this, I think returning `<script/link>` directly is more intuitive from a plugin standpoint, since we represent html in html and not in json. This way the author can also add other variables to the tag as well easily, e.g. `<link prefetch ... />`.

Reply:
Ah, but now the user has to use buildScript if they want an external link.

So they can't add other attributes like `<link prefetch href = "externa.css" />`.

The first implementation is probably better

Reply:
Good point, reverted that. ( `getResolvedAssetElement` is able to stay in `pluginUtils` though )